File: 1608.cpp
struct Solution {
    inline int specialArray(vector<int>& nums) {
        sort(nums.begin(), nums.end());
        int n = nums.size(), idx = 0;
        for (int i = 0; i <= n; ++i){
            while (idx < n && nums[idx] < i){
                ++idx;
            }
            if (idx == n - i){
                return i;
            }
        }
        return -1;
    }
};

File: sol/1608.py
class Solution:
    def specialArray(self, nums: List[int]) -> int:
        n = len(nums)
        nums.sort()
        idx = 0
        for i in range(n + 1):
            while idx < n and nums[idx] < i:
                idx += 1
            if n - idx == i:
                return i
        return -1
